src/backend/src/services/teams.services.ts

```diff
@@ -284,4 +284,38 @@ export default class TeamsService {
 
     return teamTransformer(updateTeam);
   }
+
+  /**
+   * @param submitter a user who's archiving the team
+   * @param teamId a id of team to be updated
+   * @returns the archived team
+   * @throws if the team is not found, the submitter has no privilege, the team has any projects that are not complete
+   */
+  static async archiveTeam(subimtter: User, teamId: string): Promise<Team> {
+    const team = await prisma.team.findFirst({
+      where: { teamId },
+      ...teamQueryArgs
+    });
+
+    if (!team) throw new NotFoundException('Team', teamId);
+
+    if (!isAdmin(subimtter.role)) {
+      throw new AccessDeniedException('You must be an admin or above to archive a team');
+    }
+
+    if (team.projects.some((project) => project.wbsElement.status !== WBS_Element_Status.COMPLETE)) {
+      throw new HttpException(400, 'A team is not archivable if it has any active projects, or incomplete projects');
+    }
+
+    const updatedTeam = await prisma.team.update({
+      where: { teamId },
+      ...teamQueryArgs,
+      data: {
+        userArchivedId: subimtter.userId,
+        dateArchived: team.dateArchived !== null ? new Date() : null
```

Comment:
oh right, is this one where we archive/ unarchive with the same endpoint? if so you should put that in the comments somewhere